src/lib/operators/aggregate_sort.cpp

```diff
@@ -256,39 +282,99 @@ void AggregateSort::_set_and_write_aggregate_value(
   }
 }
 
+Segments AggregateSort::_get_segments_of_chunk(const std::shared_ptr<const Table>& input_table,
+                                               const ChunkID chunk_id) {
+  Segments segments{};
+  const auto column_count = input_table->column_count();
+  segments.reserve(column_count);
+  const auto chunk = input_table->get_chunk(chunk_id);
+
+  for (auto column_id = ColumnID{0}; column_id < column_count; ++column_id) {
+    segments.emplace_back(chunk->get_segment(column_id));
+  }
+  return segments;
+}
+
+std::shared_ptr<Table> AggregateSort::_sort_table_chunk_wise(const std::shared_ptr<const Table>& input_table,
+                                                             const std::vector<ColumnID>& groupby_column_ids) {
+  auto output_table =
+      std::make_shared<Table>(input_table->column_definitions(), TableType::References, std::nullopt, UseMvcc::No);
+
+  const auto chunk_count = input_table->chunk_count();
+  for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
+    if (!input_table->get_chunk(chunk_id)) continue;
+
+    const auto chunk = std::make_shared<Chunk>(_get_segments_of_chunk(input_table, chunk_id));
+    // const auto& chunk = input_table->get_chunk(chunk_id);  // TODO: why not?
```

Comment:
How was this not catched by the linter? :(

Reply:
Hmm, good point. Does it complain if you remove the two spaces in front of `  // TODO`?

Reply:
Changed what I wanted to change when I added the todo. Moved the table creation to the else part.
I cannot simply forward a chunk to a new table, since we have an `std::shared_ptr<const Table>` and the constructor takes a pointer to a non-const table. I am not super happy with first extracting the segments and then passing them to `append_chunk()`, but I don't want to change the(/add a) constructor either. Your opinion?
 2afa612981a6824c04596c91cc4311ddbc0a4729

Reply:
Wouldn't touch the constness either.